crates/net/downloaders/src/headers/reverse_headers.rs

```diff
@@ -333,6 +333,7 @@ where
                     .count();
                 // removes all headers that are higher than current target
                 self.queued_validated_headers.drain(..skip);
+                self.queued_validated_headers.shrink_to_fit();
```

Comment:
I don't think this is necessary here, because this merely removes a few headers from the front, and we will go through `next_batch` eventually